src/main.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"fmt"
+	"html/template"
+	"net/http"
+	"os"
+	"time"
+)
+
+func loadHtmlTemplates() (*template.Template, error) {
+	if err := os.Chdir("src/templates"); err != nil {
+		return nil, err
+	}
+	defer os.Chdir("../..")
+
+	tmpl := template.New("HTML templates")
+	return tmpl.ParseFiles("index.html", "secret_page.html")
+}
+
+func main() {
+	tmpl, err := loadHtmlTemplates()
+	if err != nil {
+		panic(err)
+	}
+
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		dt := time.Now()
+		hour, min, sec := dt.Clock()
+		dtStr := fmt.Sprintf("%d:%d:%d", hour, min, sec)
+
+		if err := tmpl.ExecuteTemplate(w, "/", dtStr); err != nil {
+			panic(err)
+		}
+	})
+
+	http.HandleFunc("/secret_page", func(w http.ResponseWriter, r *http.Request) {
+		if err := tmpl.ExecuteTemplate(w, "Secret Page", nil); err != nil {
+			panic(err)
+		}
+	})
```

Comment:
Should I move this handler to a separate function then? (if there is some kind of a style requirement, this should be moved of course but I find this handler pretty short)

Reply:
I guess it should be separate. And don't forget to recover panic, pls.